gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/testing/LocalDatastoreHelper.java

```diff
@@ -589,80 +539,100 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IO
+      return tempSocket.getLocalPort();
+    } catch (IOException e) {
+      return -1;
     }
   }
 
-  private static Map<String, String> parseArgs(String[] args) {
-    Map<String, String> parsedArgs = new HashMap<String, String>();
-    for (String arg : args) {
-      if (arg.startsWith("--port=")) {
-        parsedArgs.put("port", arg.substring("--port=".length()));
-      } else if (arg.equals("START") || arg.equals("STOP")) {
-        parsedArgs.put("action", arg);
-      } else {
-        throw new RuntimeException("Only accepts START, STOP, and --port=<port #> as arguments");
-      }
-    }
-    if (parsedArgs.get("action") == null) {
-      throw new RuntimeException("EXPECTING START | STOP");
-    }
-    return parsedArgs;
+  /**
+   * Returns a {@link DatastoreOptions} instance that sets the host to use the Datastore emulator
+   * on localhost.
+   */
+  public DatastoreOptions options() {
+    return DatastoreOptions.builder()
+        .projectId(projectId)
+        .host("http://localhost:" + Integer.toString(port))
+        .authCredentials(AuthCredentials.noAuth())
+        .build();
   }
 
-  public static boolean isActive(String projectId, int port) {
-    try {
-      StringBuilder urlBuilder = new StringBuilder("http://localhost:").append(port);
-      urlBuilder.append("/datastore/v1beta2/datasets/").append(projectId).append("/lookup");
-      URL url = new URL(urlBuilder.toString());
-      try (BufferedReader reader =
-          new BufferedReader(new InputStreamReader(url.openStream(), UTF_8))) {
-        return "Valid RPC".equals(reader.readLine());
-      }
-    } catch (IOException ignore) {
-      // assume not active
-      return false;
+  /**
+   * Returns the project ID associated with this local Datastore emulator.
+   */
+  public String projectId() {
+    return projectId;
+  }
+
+  /**
+   * Returns the port on localhost to which the local Datastore listens for requests.
+   */
+  public int port() {
+    return port;
+  }
+
+  /**
+   * Returns the consistency setting for the local Datastore emulator.
+   */
+  public double consistency() {
+    return consistency;
+  }
+
+  /**
+   * Starts the local Datastore for the specific project.
+   *
+   * This will unzip the gcd tool, create the project and start it. All content is written to a
+   * temporary directory that will be deleted when {@link #stop()} is called (or when the program
+   * terminates) to make sure that no left-over data from prior runs is used.
+   *
+   * @param consistency the fraction of Datastore writes that are immediately visible to global
+   *     queries, with 0.0 resulting in no attempts succeeding, and 1.0 resulting in all attempts succeeding. Note
+   *     that setting this to 1.0 may mask incorrect assumptions about the consistency of
+   *     non-ancestor queries; non-ancestor queries are eventually consistent.
+   */
+  public static LocalDatastoreHelper start(String projectId, double consistency)
```

Comment:
We could consider taking out the `findAndStartGcd()` call and renaming this (and the next) method to `create`. `findAndStartGcd()` could be renamed to `start()` and become public. This should be slightly more similar to remote helpers.

Also I think there might be a value in providing a way of setting the port if this class is ever used to start a remote emulator (I am thinking about firewalls here).